openmeter/credit/balance.go

```diff
@@ -37,44 +36,45 @@ type BalanceHistoryParams struct {
 var _ BalanceConnector = &connector{}
 
 func (m *connector) GetBalanceOfOwner(ctx context.Context, owner grant.NamespacedOwner, at time.Time) (*balance.Snapshot, error) {
```

Comment:
As far as I see the `GetBalanceOfOwner` and `GetBalanceHistoryOfOwner` is the same up to the point where we execute the engine. Can we merge that part into a function and just invoke different transformations on the engine results?